internal/oci/runtime_oci.go

```diff
@@ -1101,29 +1101,38 @@ func (c *Container) conmonPidFilePath() string {
 // It returns immediately with a channel to find when path had one of those events.
 // done can be used to stop the watch.
 // WatchForFile is responsible for closing all internal channels and the returned channel, but not for closing done.
-func WatchForFile(path string, done chan struct{}, opsToWatch ...notify.Event) (chan struct{}, error) {
-	eiCh := make(chan notify.EventInfo, 1)
+func WatchForFile(path string, done chan struct{}, opsToWatch ...fsnotify.Op) (chan struct{}, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
 	ch := make(chan struct{})
 
 	dir := filepath.Dir(path)
-	if err := notify.Watch(dir, eiCh, opsToWatch...); err != nil {
-		return nil, err
-	}
 	go func() {
+		defer watcher.Close()
```

Comment:
This could be put after the err check on NewWatcher.

Reply:
I leave it here because there exists the chance we'll want to add a done channel in this goroutine (it does not seem to be needed now). Leaving it in the goroutine allows us just to extend the for loop, not move around the channels. If that isn't clear though I could document/change it. wydt